app/src/main/java/com/example/androidthings/assistant/AssistantActivity.java

```diff
@@ -213,7 +213,9 @@ public void onAssistantResponse(final String response) {
                         mMainHandler.post(new Runnable() {
                             @Override
                             public void run() {
-                                mAssistantRequestsAdapter.add("Google Assistant: " + response);
+                                if(response != ""){
```

Comment:
The Java `String` class has a useful `isEmpty` utility that you may want to use instead. Also, it may be useful to not execute the runnable at all if the string is empty.